jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactoryInternals.java

```diff
@@ -52,6 +52,9 @@
         PathResourceFactory pathResourceFactory = new PathResourceFactory();
         RESOURCE_FACTORIES.put("file", pathResourceFactory);
         RESOURCE_FACTORIES.put("jrt", pathResourceFactory);
+
+        if (GraalIssue5720PathResourceFactory.ENABLE_NATIVE_IMAGE_RESOURCE_SCHEME)
+            RESOURCE_FACTORIES.put("resource", new GraalIssue5720PathResourceFactory());
```

Comment:
This whole block can be replaced with a more general approach, that also supports the bug workaround ...

``` java
        // The default resource factories
        MountedPathResourceFactory mountedPathResourceFactory = new MountedPathResourceFactory();
        RESOURCE_FACTORIES.put("jar", mountedPathResourceFactory);
        PathResourceFactory pathResourceFactory = new PathResourceFactory();
        RESOURCE_FACTORIES.put("file", pathResourceFactory);
        RESOURCE_FACTORIES.put("jrt", pathResourceFactory);

        URL url = ResourceFactoryInternals.class.getResource("/org/eclipse/jetty/version/build.properties");
        if (url != null)
        {
            if (GraalIssue5720PathResourceFactory.hasBug(url))
                RESOURCE_FACTORIES.put(url.getProtocol(), new GraalIssue5720PathResourceFactory());
            else if (!RESOURCE_FACTORIES.contains(url.getProtocol()))
                RESOURCE_FACTORIES.put(url.getProtocol(), mountedPathResourceFactory);
        }
```

Reply:
The `GraalIssue5720PathResourceFactory.hasBug(URL)` would need to check for protocol `resource`, and then perform a path filesystem mount and test of `Path.toURI()` to detect the bug.
Thankfully, this static list is only initialized once at the start of the JVM, so it's not a performance hit.

Reply:
I actually don't see a reason for `MountedPathResourceFactory` to exist. Maybe I'm missing a point but you could do this all with a single `PathResourceFactory`. All you'd need to do is to wrap `Path.of` (or `Paths.get`) with a `try-catch` block to find `FileSystemNotFoundException` very much like I do [in this demo code](https://github.com/joakime/graalvm-experiments/blob/ck/src/main/java/graaldemo/Demo.java). You don't have to manually wrangle with exclamation points for jar URLs. Just create a `newFileSystem` for the requested URI. It will work just fine.

Such `PathResourceFactory` would be capable of supporting all `Path`-providing URL schemes.
It would still be the place where would need to handle the "graal issue 5720"-specific manipulation, which is how it was in my original solution.

Reply:
If `GraalIssue5720PathResourceFactory` and `GraalIssue5720PathResource` classes only exist because of the graal bug && line 56 will be deleted once it's fixed && and line 57 can be moved elsewhere from `ResourceFactoryInternals` once it's fixed, then I'm ok with it. 

What I would like us to do is to use this example to learn how to handle, in a generic way, any url scheme. So far we've now had 2 examples: `bundleresource:` and `resource:`. Have we used those to extract a meaningful generic mechanism? If so, can we document that please.